database-commons/src/main/java/io/cdap/plugin/db/batch/sink/AbstractDBSink.java

```diff
@@ -211,13 +211,21 @@ public void transform(StructuredRecord input, Emitter<KeyValue<DBRecord, NullWri
       outputFields.add(field);
     }
     StructuredRecord.Builder output = StructuredRecord.builder(outputSchema);
-    for (String column : columns) {
-      output.set(column, input.get(column));
+    for (int i = 0; i < columns.size(); i++) {
+      transformField(input, i, output);
```

Comment:
Albert said on last sprint review that we should not transform field, instead we should fail writing field if it's not compatible with output schema

Reply:
Since all Oracle numeric types are based on NUMBER(i.e. INTEGER type is actually NUMBER(38, 0)) we won't be able to use Oracle Sink Plugin with other sources.
It's safe to write primitives as values of decimal logical type in the case of valid precision.

Reply:
Removed.